tutorials/git-hooks.md

```diff
@@ -0,0 +1,50 @@
+# Czym są git hooki
+
+Przepływ operacji w gitcie jest dość dobrze znany i ustalony. Wykonujemy kolejno operacje `git add`, `git commit` `git pull` `git push`, etc. Co jednak jeśli chcielibyśmy aby w pewnych sytuacjach działo się coś dodatkowego.
+
+Okazuje się, że taka potrzeba zdarza się nader często w większych projektach. Np. jeśli byśmy chcieli przy każdym commit sprawdzić czy kod przechodzi testy jednostkowe. Okazuje się, że odpowiedzią są na to 
+tzw. git hooki. 
+
+W każdym repozytorium wyciągniętym lokalnie można odszukać ukryty katalog `.git` gdzie przechowywane są dane o commitach i wersjach. Wśród nich znajduje się informacja o hookach
+
+```sh
+ls .git/hooks/
+applypatch-msg.sample      post-update.sample     pre-merge-commit.sample    pre-receive.sample
+commit-msg.sample          pre-applypatch.sample  prepare-commit-msg.sample  push-to-checkout.sample
+fsmonitor-watchman.sample  pre-push.sample        update.sample              pre-commit.sample      pre-rebase.sample
+```
+
+Każdy z plików sample to tylko szablon jak dany hook mógłby wyglądać + opis działania. Każdemu hookowi jest przypisana odpowiednia akcja. Najpopularniejszym hookiem jest pre-commit - który jak nazwa wskazuje wykona się przy akcji commit - ale poprzedzając sam commit. 
+
+Aby używać konkretnego hooka należy wykonać 3 rzeczy
+
+* (prosta) utworzyć plik o nazwie hooka ale już bez .sample czy innego rozszerzenia
+* (prosta) nadać mu uprawnienia wykonywania `chmod +x hook`
```

Comment:
gdzie należy to zrobić, w terminalu w pliku w jakiś sposób, w notatniku?

Reply:
dopisuje